File: javascript/bob/bob.js
export const hey = (statement) => {
  let answer = ''

  switch(statementType(statement.trim())) {
    case 'question':
      answer = 'Sure.'
      break
    case 'shouting':
      answer = 'Whoa, chill out!'
      break
    case 'forceful question':
      answer = "Calm down, I know what I'm doing!"
      break
    case 'silence':
      answer = 'Fine. Be that way!'
      break
    default:
      answer = 'Whatever.'
  }

  return answer;
}


const statementType = (statement) => {
  if (isAQuestion(statement) && isShouting(statement)) {
    return 'forceful question'
  };

  if (isAQuestion(statement)) {
    return 'question'
  };

  if (isShouting(statement)) {
    return 'shouting'
  };

  if (isEmpty(statement)) {
    return 'silence'
  };
}

const isAQuestion = (statement) => {
  return (statement.slice(-1) == '?')
}

const isShouting = (statement) => {
  return ((statement.toUpperCase() === statement) && containsSomeLetters(statement));
}

const containsSomeLetters = (statement) => {
    return (/[a-z]/i.test(statement));
};

const isEmpty = (statement) => {
    return (statement.replace(/\s/g,'') == '');
};

File: ruby/bob/bob.rb
class Bob
  def self.hey remark
    remark = remark.strip
    
    if remark.chars.last == "?" && remark.upcase == remark && remark.upcase.scan(/[A-Z]/).length != 0
      "Calm down, I know what I'm doing!"
    elsif remark.chars.last == "?"
      "Sure."
    elsif remark.upcase == remark && remark.upcase.scan(/[A-Z]/).length != 0
      "Whoa, chill out!"
    elsif remark.length == remark.scan(/\s/).length
      "Fine. Be that way!"
    else
      "Whatever."
    end
  end
end
